src/index.js

```diff
@@ -249,6 +250,102 @@ exports.import = (target, dagService, options, callback) => {
   // function streamImporter (stream, callback) {}
 }
 
-exports.export = function () {
-  // export into files by hash
+exports.export = function (hash, dagService, options, callback) {
+  if (typeof options === 'function') { callback = options; options = {} }
+  const ee = new events.EventEmitter()
+  dagService.get(hash, (err, fetchedNode) => {
+    if (err) {
+      return err
+    }
+    const data = UnixFS.unmarshal(fetchedNode.data)
+    const type = data.type
+    if (type === 'directory') {
+      dirExporter(fetchedNode, hash, callback)
+    }
+    if (type === 'file') {
+      fileExporter(fetchedNode, hash, false, callback)
+    }
+  })
+  return ee
+
+  function fileExporter (node, name, dir, callback) {
```

Comment:
I think it would be nice if `fileExporter` and `dirExporter` were in separate files and get the event emitter passed as an argument, that would make testing them independently also easier.